### PR TITLE
T11000

### PR DESCRIPTION
**Pull request by mattdangerw**
NOT FOR MERGING! This is just to see and review the work on top of the gnome release base.

Our master will be hard reset to this branch when ready.

https://phabricator.endlessm.com/T11000

### CODE REVIEW

File: gtk/gtkcssenumvalueprivate.h [hunk 1]
```diff
@@ -34,7 +34,8 @@ GtkBorderStyle  _gtk_css_border_style_value_get       (const GtkCssValue *value)
 GtkCssValue *   _gtk_css_font_size_value_new          (GtkCssFontSize     size);
 GtkCssValue *   _gtk_css_font_size_value_try_parse    (GtkCssParser      *parser);
 GtkCssFontSize  _gtk_css_font_size_value_get          (const GtkCssValue *value);
-double          _gtk_css_font_size_get_default        (GtkStyleProviderPrivate *provider);
+double          gtk_css_font_size_get_default_px      (GtkStyleProviderPrivate *provider,
```
Comment (ptomato) [inline gtk/gtkcssenumvalueprivate.h]: Is this becoming public?

Comment (mattdangerw) [inline gtk/gtkcssenumvalueprivate.h]: @ptomato nope, but private functions are not being marked with underscores, just kept in private header files.

I guess so you don't have to do a massive find a replace every time you make private api public? I think the underscore was used in some old makefile rule that's not needed anymore.

**Comment by mattdangerw** [comment]
@cosimoc the last commit in this change tries to fix up our custom css for the new gtk.

The app overrides are a bit tricky, a lot of that styling is done by selecting based on GTypeName, which won't work anymore. I'm not sure if we plan on bumping sushi, gedit, evince, and gcalctool to the gnome 3.20 versions, and if we do if they even bother to set css names that we can style.

Should I try to fixup this styling overrides so they don't rely on gtype name at all? Then maybe things will work with both 3.18 and 3.20 gnome apps? If you have a chance, lemme know what you think the best approach is here.

**Comment by cosimoc** [comment]
@mattdangerw To be honest I'm not really sure why we have those overrides at all, or if they even still work these days...
Here's how I would approach it:
- remove all the app-specific overrides
- keep the WM decoration and mutter styling
- it's unclear to me why the "Endless window top bar" styling is here instead of eos-sdk. If I understand correctly what that's supposed to select, I would move it to eos-sdk's CSS file instead

Let me know if this makes sense - the rest of the PR looks good to me. As a side note, it may even make sense to try and have some of those commits upstream. I will try to comment on those - maybe you can open upstream issues and attach the patches there? (not a blocker for the PR of course!)

**Comment by mattdangerw** [comment]
@cosimoc all revised.

Still have the damn debian build failures that are driving me insane. reftests module loading seems is behaving strangely, but only in a obs chroot. Don't merge yet

Anyway basically remove everything from our custom file except the topbar theme. Talking with @magcius decided to pull in the mutter commits updating the theme (basically everything that touches theme.c), so put those up in a separate PR.

Then did a little cleanup of the styling, think we shifted the appearance slightly in the mutter rebase. But take a look and lemme know if any of it was intentional.

**Comment by cosimoc** [comment]
@mattdangerw Looks all good to me now. So I guess the only real blocker at this point are the test failures? I would not mind merging it with the tests commented out in the meantime and open a follow-up story.

**Comment by mattdangerw** [comment]
@cosimoc sounds good splitting out. Updating the tmp commit to just drop the tests from the makefile will should leave them out of the dist tarball and keep them from getting to obs. The tests run are run in make by invoking gtk-reftest which just runs on every `*.ui` in the directory listing, so there's no easy way to comment them out....

**Comment by mattdangerw** [comment]
https://phabricator.endlessm.com/T11411

**Comment by mattdangerw** [comment]
Closing this was just for review

**Pull request closed without merging**